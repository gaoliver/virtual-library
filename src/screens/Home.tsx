import React, {useState} from 'react';
import {AspectRatio, Center, Image} from 'native-base';
import {NativeStackScreenProps} from '@react-navigation/native-stack';
import {SearchBar} from '@/components';
import {RootStackParamList} from 'App';

type HomeProps = NativeStackScreenProps<RootStackParamList, 'Home'>;

export const Home: React.FC<HomeProps> = ({navigation}) => {
  const [searchQuery, setSearchQuery] = useState('');

  const logo = require('@assets/images/Virtual_Library-logos_white.png');

  const handleSearch = () => {
    navigation.navigate('SearchResults', {
      query: searchQuery,
    });

    setSearchQuery('');
  };

  return (
    <Center safeArea flex={1} bgColor="secondary" px={12}>
      <AspectRatio ratio={30 / 17} w="200px">
        <Image
          w="100%"
          h="100%"
          source={logo}
          alt="Virutal Library logo"
          testID="virtual-library-logo"
        />
      </AspectRatio>

      <SearchBar
        mt={4}
        value={searchQuery}
        onChangeText={setSearchQuery}
        onBlur={handleSearch}
        testID="searchbar"
      />
    </Center>
  );
};
