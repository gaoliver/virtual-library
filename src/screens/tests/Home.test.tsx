import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import {Home} from '../Home';
import {NativeBaseProvider} from 'native-base';

const navigation = {
  navigate: jest.fn(),
};

const inset = {
  frame: {x: 0, y: 0, width: 0, height: 0},
  insets: {top: 0, left: 0, right: 0, bottom: 0},
};

describe('Home Screen', () => {
  it('renders correctly', () => {
    const {toJSON} = render(
      <NativeBaseProvider>
        <Home navigation={navigation} />
      </NativeBaseProvider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('navigates to SearchResults on search', () => {
    const {getByTestId} = render(
      <NativeBaseProvider initialWindowMetrics={inset}>
        <Home navigation={navigation} />
      </NativeBaseProvider>,
    );
    const searchInput = getByTestId('searchbar');
    fireEvent.changeText(searchInput, 'Search Query');
    fireEvent(searchInput, 'blur');

    expect(navigation.navigate).toHaveBeenCalledWith('SearchResults', {
      query: 'Search Query',
    });
  });
});
