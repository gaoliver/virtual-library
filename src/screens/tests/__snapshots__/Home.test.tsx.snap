// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
